### PR TITLE
test(tree2): update state-based axiom test harness to generate appropriate rebase metadata

### PR DESCRIPTION
**Pull request by yann-achard-MS**
Updates the state-based axiom test harness to generate appropriate rebase metadata (needed by sequence field).
Refactors some the state-based axiom test harness functions to reduce the number of nested scopes.

Also cuts down the set of generated edits to a smaller set to facilitate debugging failures.

### CODE REVIEW

**Review by alex-pardes — COMMENTED** [review]
The description says "to facilitates" instead of "to facilitate."

File: experimental/dds/tree2/src/test/rebaserAxiomaticTests.ts [hunk 1]
```diff
@@ -48,11 +50,15 @@ export function runExhaustiveComposeRebaseSuite<TContent, TChangeset>(
...
 	): TaggedChange<TChangeset> {
 		let currChange = change;
-		const metadata = defaultRevisionMetadataFromChanges([change, ...baseChanges]);
-		for (const base of baseChanges) {
+		const metadata = rebaseRevisionMetadataFromInfo(
+			defaultRevInfosFromChanges(rebasePath),
+			baseBranch.map(({ revision }) => revision),
+		);
+		for (const base of rebasePath) {
 			currChange = tagChange(rebase(currChange.change, base, metadata), currChange.revision);
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/test/rebaserAxiomaticTests.ts:62]: We have to create a separate `RebaseRevisionMetadata` for each step of the rebase. See example here: https://github.com/microsoft/FluidFramework/blob/ea326458fe2ce16d746149ecc220447beb86cdd8/experimental/dds/tree2/src/test/feature-libraries/sequence-field/utils.ts#L141

File: experimental/dds/tree2/src/test/rebaserAxiomaticTests.ts [hunk 2]
```diff
@@ -225,37 +232,50 @@ export function runExhaustiveComposeRebaseSuite<TContent, TChangeset>(
...
 								const edit = sourceEdits[i];
-								const editsToRebaseOver = [
+								const rebasePath = [
 									...inverses.slice(sourceEdits.length - i),
 									editToRebaseOver,
 									...rebasedEditsWithoutCompose,
 								];
 								rebasedEditsWithoutCompose.push(
-									rebaseTagged(edit, ...editsToRebaseOver),
+									rebaseTagged(edit, rebasePath, [editToRebaseOver]),
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/test/rebaserAxiomaticTests.ts:241]: I'm confused about what `rebaseTagged` is supposed to do.

Comment (yann-achard-MS) [reply experimental/dds/tree2/src/test/rebaserAxiomaticTests.ts:241]: The same thing that [rebaseOverChanges](https://github.com/microsoft/FluidFramework/blob/main/experimental/dds/tree2/src/test/feature-libraries/sequence-field/utils.ts#L141-L160) is, but more generic?

File: experimental/dds/tree2/src/test/rebaserAxiomaticTests.ts [hunk 3]
```diff
@@ -154,12 +157,12 @@ export function runExhaustiveComposeRebaseSuite<TContent, TChangeset>(
 							);
 							const rebaseWithoutCompose = rebaseTagged(
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/test/rebaserAxiomaticTests.ts:158]: I'm finding this test difficult to read. Can we break it up into smaller functions?

**Review by alex-pardes — APPROVED** [review]

**Pull request merged**